### ISSUE TITLE
check_files.pl: -l and -F <wildcard> with no files matching wildcard results in UNKNOWN ERROR

### ISSUE THREAD

**Issue by bradcavanagh** [willixix/naglio-plugins#36]
If I do something like:

```
check_files.pl -D /home/user/directory -F *.txt -l -c 1 -v
```

...and there are no files matching `*.txt` in `/home/user/directory`, I get the following error:

```
Translated filespec '*.txt' to regex '.*\.txt'
Only critical value check is specified - setting warning to ~
check_files.pl plugin version 0.417
Alarm at 30
Changed to directory '/home/user/directory/'
Command: LANG=C ls -l
Executing LANG=C ls -l *.txt 2>&1
got line: ls: cannot access *.txt: No such file or directory
    parsed: type=unset
UNKNOWN ERROR - execution of LANG=C ls -l *.txt resulted in an error 512 - 
```

I'm using `-l` because I have a lot of files in the directory, and the Nagios check is timing out if I leave it off. I could increase the timeout, but adding the `-l` seems to be the better solution.

I'll have a go at patching it, but can't make any promises it'll be a good patch. :-)

**Comment by willixix** [comment]
The error coming out of this code:

if (!defined($o_stdin) && !close(SHELL_DATA)) {
    print "UNKNOWN ERROR - execution of $shell_command resulted in an error $? - $!";
    exit $ERRORS{'UNKNOWN'};
}

This has an issue because it makes reference to object that is now opened and named in a function, so In above close(SHELL_DATA) should be replaced with close($READTHIS). However I have a feeling this would still result in the same error. Try and let me know.

**Comment by bradcavanagh** [comment]
Yep, same error. The root cause is actually from `ls`. If you just run `ls -l *.txt` on the command-line in a directory that has no `*.txt` files, `ls` returns a status code of 2 (serious error) instead of 0 (success). I don't know of any way to get it to not do this, and I don't think you'd want to patch your code to silently accept an error code from `ls`...

**Comment by willixix** [comment]
All right I fixed this and pushed new version (0.42 version of plugin). I simply check for "No such file or directory" as a special case

**Comment by bradcavanagh** [comment]
Awesome! I've installed the new version and verify that this patch fixes this issue.
